src/utils/entitySpec.js

```diff
@@ -15,6 +15,26 @@ const toLowerCaseMap = {
   mavencentralsource: NONE
 }
 
+const NPM_WEBSITE = 'npmjs.com'
+const GITHUB_WEBSITE = 'github.com'
+const MAVEN_WEBSITE = 'mvnrepository.com'
+const NUGET_WEBSITE = 'nuget.org'
+const PYPI_WEBSITE = 'pypi.org'
+const RUBYGEM_WEBSITE = 'rubygems.org'
+
+const providerPath = {
```

Comment:
These are the defaults but not necessarily always true. For example, we should support npms coming from github repos (we actually use this in clearlydefined itself) and release zips coming from GitHub. So the provider function that teases apart the dropped url should be the one that forms the path.

Reply:
I think this could a be a new requirement. Can we address this in a new specific issue?

Reply:
Its not new, the system design has always been this way. That is why we have type and provider as separate concepts. We've not done a lot of instances of it but would like to avoid taking implementation approaches that would prevent that. I'm fine getting something working in a simple form. I think the suggestion above to have an object that captures the behavior of a particular provider and/or type would address this.